crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1483,8 +1508,19 @@ where
             return Ok(())
         }
 
+        let sync_target_state = match self.forkchoice_state_tracker.sync_target_state() {
+            Some(current_state) => current_state,
+            None => {
+                // This is only possible if the node was run with `debug.tip`
+                // argument and without CL.
+                warn!(target: "consensus::engine", "No fork choice state available");
+                return Ok(())
+            }
+        };
+
         // update the canon chain if continuous is enabled
-        if self.sync.run_pipeline_continuously() {
+        if self.sync.run_pipeline_continuously() || sync_target_state.finalized_block_hash.is_zero()
+        {
```

Comment:
this might break continuous syncing. previously, we first checked if it's continuous and only then looked at the forkchoice state which iirc is not set during continuous

Reply:
fixed